### ISSUE TITLE
Add nice 404 page

### ISSUE THREAD

**Issue by Tom-van-Woudenberg** [TeachBooks/deploy-book-workflow#45]
To be compatible with: https://github.com/TeachBooks/deploy-book-workflow/pull/39
@severinbratus, am I correct that the current 404 page is described here: https://github.com/TeachBooks/deploy-book-workflow/blob/dc47d34ae3211fad64b30e55661162d923729551/.github/workflows/deploy-book.yml#L463 and that the current [404.html](https://github.com/TeachBooks/deploy-book-workflow/blob/main/404.html) and [404.md](https://github.com/TeachBooks/deploy-book-workflow/blob/main/404.md) can be deleted?
Is it possible to use this default 404 page only if no custom 404 page is provided on the root of the repo to be in line with the [github documentation](https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-custom-404-page-for-your-github-pages-site)?

**Comment by severinbratus** [comment]
Yes, that is correct. Yes, this should be possible. I see two tasks to do here:

- [ ] Make a better default 404 page (e.g. also with a list of deployed branches, or a link to the latest successful summary)
- [ ] Use the 404.html in the root dir if provided

**Comment by AleksSliw** [comment]
I am editing the template for the generated 404 page and a question popped up. What should be the expected behaviour for 404 redirect? inputting a wrong URL in the form of  ```valid-repository/wrong-name``` will redirect to ```valid-repository/main/wrong-name``` (which is still invalid and will show 404), while inputting ```valid-repository/right-branch/wrong-name``` will just show the 404 page with that same URL.

Im assuming the idea is to redirect to the main branch if the repository exists?

**Comment by Tom-van-Woudenberg** [comment]
I think the 404 page should never redirect, only show a button to bring you to the root of valid-repository
There'll always be a page on root.

**Comment by Tom-van-Woudenberg** [comment]
Maybe we can make the 404 page from the new WordPress website? In that theme

**Comment by Tom-van-Woudenberg** [comment]
Partly done in #93
